client/client.go

```diff
@@ -122,25 +123,41 @@ func bind(cfg *Config, conn ldaputil.Connection) error {
 		return errors.New("unable to bind due to lack of configured password")
 	}
 
-	if cfg.BindDN == "" {
-		return errors.New("must provide binddn")
-	}
-
-	origErr := conn.Bind(cfg.BindDN, cfg.BindPassword)
-	if origErr == nil {
+	if cfg.UPNDomain != "" {
+		origErr := conn.Bind(fmt.Sprintf("%s@%s", ldaputil.EscapeLDAPValue(cfg.BindDN), cfg.UPNDomain), cfg.BindPassword)
```

Comment:
I would've expected that we escape the `UPNDomain` as well. I'm hesitant to change the behavior at this point because we don't escape it in both LDAP auth ([client.go#L140-L178](https://github.com/hashicorp/vault/blob/main/sdk/helper/ldaputil/client.go#L140-L178)) and AD secrets engine ([client.go#L128-L141](https://github.com/hashicorp/vault-plugin-secrets-ad/blob/main/plugin/client/client.go#L128-L141)).

Reply:
:+1: